### PR TITLE
PHOENIX-5721 Unify 4.x branches (for HBase 1.5, 1.4 as well as 1.3)

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DelegateRegionCoprocessorEnvironment.java [hunk 1]
```diff
@@ -40,10 +40,9 @@
  * clone the configuration provided by the HBase coprocessor environment before modifying it. So
  * this class comes in handy where we have to return our custom config.
  */
-public class DelegateRegionCoprocessorEnvironment implements RegionCoprocessorEnvironment {
+public class DelegateRegionCoprocessorEnvironment extends  CompatDelegateRegionCoprocessorEnvironment {
```
Comment (joshelser) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DelegateRegionCoprocessorEnvironment.java]: Nit: extra space

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DelegateRegionObserver.java [hunk 2]
```diff
@@ -676,40 +675,13 @@ public Cell postMutationBeforeWAL(ObserverContext<RegionCoprocessorEnvironment>
     public DeleteTracker postInstantiateDeleteTracker(
             ObserverContext<RegionCoprocessorEnvironment> ctx, DeleteTracker delTracker)
             throws IOException {
-        return delegate.postInstantiateDeleteTracker(ctx, delTracker);
+        return   delegate.postInstantiateDeleteTracker(ctx, delTracker);
```
Comment (joshelser) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DelegateRegionObserver.java]: nit: extra space

File: phoenix-hbase-compat-1.3.0/src/main/java/org/apache/phoenix/compat/hbase/CompatPhoenixRpcScheduler.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
```
Comment (joshelser) [inline phoenix-hbase-compat-1.3.0/src/main/java/org/apache/phoenix/compat/hbase/CompatPhoenixRpcScheduler.java]: License header got a little mangled here. Make sure `mvn apache-rat:check` still passes, please.

File: phoenix-hbase-compat-1.4.0/src/main/java/org/apache/phoenix/compat/hbase/CompatPhoenixRpcScheduler.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
```
Comment (joshelser) [inline phoenix-hbase-compat-1.4.0/src/main/java/org/apache/phoenix/compat/hbase/CompatPhoenixRpcScheduler.java]: Header here, too.

File: phoenix-hbase-compat-1.4.0/src/main/java/org/apache/phoenix/monitoring/GlobalMetricRegistriesAdapter.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+import org.apache.hadoop.metrics2.lib.Interns;
+import org.apache.hadoop.metrics2.lib.MutableHistogram;
+import org.apache.hadoop.metrics2.source.JvmMetrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contents mostly copied from GlobalMetricRegistriesAdapter class from hbase-hadoop2-compat The
+ * adapter attaches HBase's MetricRegistry to Hadoop's DefaultMetricsSystem 
+ * Note: This DOES NOThandle dynamic attach/detach of registries
```
Comment (joshelser) [inline phoenix-hbase-compat-1.4.0/src/main/java/org/apache/phoenix/monitoring/GlobalMetricRegistriesAdapter.java]: nit: NOT_handle

**Comment by stoty** [comment]
Thanks for the careful review, @joshelser

**Comment by stoty** [comment]
I haven't updated the patches in the JIRA, as it seems to be down ATM.

**Pull request closed without merging**